host/src/c_api/c_api.cc

```diff
@@ -56,6 +56,7 @@ if (result != OE_OK) {                                          \
       result,                                                   \
       oe_result_str(result));                                   \
   oe_terminate_enclave(Enclave::getInstance().getEnclave());    \
+  return result;                                                \
```

Comment:
Good catch on this!

If the enclave isn't initialized or the call returns an error, do we want to instead exit the process instead of returning? If there's no enclave, it seems pointless to continue to execute the program, and if the enclave is terminated upon `result != OE_OK`, it also seems pointless to continue to execute the program.

The `safe_xgboost()` call in `c-api-demo.c` in vanilla XGBoost seems to follow a similar model to the one I mentioned. https://github.com/dmlc/xgboost/blob/master/demo/c-api/c-api-demo.c#L16

Reply:
I think this is best handled at the application level, instead of at the framework level? 
One could imagine that if the enclave fails to initialize, the application may try again (perhaps using a varying set of parameters), or decide to exit.
Which is what happens in `c-api-demo.c` as well: the application decides to terminate if the framework returns an error.

Reply:
Good though, I agree.